test/application.js

```diff
@@ -265,6 +268,23 @@ describe('Application', () => {
         sinon.assert.calledWith(trigger, 'xfc.unload');
       }));
     });
+
+    describe("#DOMContentLoaded", () => {
+      it("calls addEventListener only when DOM is ready", sinon.test(function() {
+        application.init({acls, secret, onReady});
+        const DOMContentLoaded_event = document.createEvent('Event');
+        DOMContentLoaded_event.initEvent('DOMContentLoaded', true, true);
+        document.addEventListener('DOMContentLoaded', () => console.log('mock DOMContentLoaded'));
+        document.dispatchEvent(DOMContentLoaded_event);
+
+        expect(document.body.addEventListener.calledOnce);
+      }));
+
+      it("does not call addEventListener when DOM is not ready", sinon.test(function() {
+        sinon.spy(document.body, 'addEventListener');
```

Comment:
shouldn't this test also verify it is added once DOM is ready? To my point in the code above, just attaching event to document will be simpler in impementation and to test.